### DISCUSSION TITLE
Is it unavoidable to trigger the loader function when my action function return value?

### DISCUSSION THREAD

**Discussion by carltin0315** [remix-run/react-router#10351] [Q&A]
I use the loader to fetch data from api and pass it as a props to a component with a form, when I edit some data and submit through the action, the loader will trigger to fetch the data again.

Can I avoid it?  Because All I want is just get the result using useAcctionData without any sideeffect, I just stay in the same component without any redirect or rerender the whole page, why the loader will trigger again?

**Comment by gmaxlev** [comment]
It seems to me that you need to use [shouldRevalidate](https://reactrouter.com/en/main/route/should-revalidate#shouldrevalidate)
